registration/authorize_test.go

```diff
@@ -122,3 +123,69 @@ func TestAuthorizeNode(t *testing.T) {
 		})
 	}
 }
+
+func TestAuthorizeNodeCommon_DuplicateStore(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	fileStorage, err := storeonce.New(ctx)
+	require.NoError(t, err)
+
+	_, err = rotation.RotateRootCertificates(ctx, fileStorage)
+	require.NoError(t, err)
+
+	// This happens on the node
+	nodeCreds, err := types.NewNodeCredentials(ctx, fileStorage)
+	require.NoError(t, err)
+	fetchReq, err := nodeCreds.CreateFetchNodeCredentialsRequest(ctx)
+	require.NoError(t, err)
+	keyId, err := nodeenrollment.KeyIdFromPkix(nodeCreds.CertificatePublicKeyPkix)
+	require.NoError(t, err)
+	nodePubKey, err := curve25519.X25519(nodeCreds.EncryptionPrivateKeyBytes, curve25519.Basepoint)
+	require.NoError(t, err)
+
+	structMap := map[string]interface{}{"foo": "bar"}
+	state, err := structpb.NewStruct(structMap)
+	require.NoError(t, err)
+
+	// Add in node information to storage so we have a key to use
+	nodeInfo := &types.NodeInformation{
+		Id:                       keyId,
+		CertificatePublicKeyPkix: nodeCreds.CertificatePublicKeyPkix,
+		CertificatePublicKeyType: nodeCreds.CertificatePrivateKeyType,
+		EncryptionPublicKeyBytes: nodePubKey,
+		EncryptionPublicKeyType:  nodeCreds.EncryptionPrivateKeyType,
+		RegistrationNonce:        nodeCreds.RegistrationNonce,
+		State:                    state,
+	}
+
+	fetchInfo, _ := registration.ValidateFetchRequestCommon(ctx, fileStorage, fetchReq)
+	_, err = registration.AuthorizeNodeCommon(ctx, fileStorage, fetchInfo)
+	require.NoError(t, err)
+
+	checkNodeInfo := &types.NodeInformation{Id: nodeInfo.Id}
+	require.NoError(t, fileStorage.Load(ctx, checkNodeInfo))
+	require.NotNil(t, checkNodeInfo)
+	assert.Equal(t, nodeInfo.Id, checkNodeInfo.Id)
+	assert.NotEmpty(t, checkNodeInfo.CertificatePublicKeyPkix)
+	assert.Equal(t, types.KEYTYPE_ED25519, checkNodeInfo.CertificatePublicKeyType)
+	assert.Len(t, checkNodeInfo.CertificateBundles, 2)
+	for _, bundle := range checkNodeInfo.CertificateBundles {
+		assert.NotEmpty(t, bundle.CertificateDer)
+		assert.NotEmpty(t, bundle.CaCertificateDer)
+		assert.NoError(t, bundle.CertificateNotBefore.CheckValid())
+		assert.False(t, bundle.CertificateNotBefore.AsTime().IsZero())
+		assert.NoError(t, bundle.CertificateNotAfter.CheckValid())
+		assert.False(t, bundle.CertificateNotAfter.AsTime().IsZero())
+	}
+	assert.NotEmpty(t, checkNodeInfo.EncryptionPublicKeyBytes)
+	assert.Equal(t, types.KEYTYPE_X25519, checkNodeInfo.EncryptionPublicKeyType)
+	assert.NotEmpty(t, checkNodeInfo.ServerEncryptionPrivateKeyBytes)
+	assert.Equal(t, types.KEYTYPE_X25519, checkNodeInfo.ServerEncryptionPrivateKeyType)
+	assert.Len(t, checkNodeInfo.RegistrationNonce, nodeenrollment.NonceSize)
+
+	// Simulate a withWrapper case where we might hit authorizeNodeCommon a second time
+	returnedNodeInfo, err := registration.AuthorizeNodeCommon(ctx, fileStorage, fetchInfo)
+	require.NoError(t, err)
+	require.Equal(t, checkNodeInfo, returnedNodeInfo)
```

Comment:
I'm not sure `require.Equal` works properly with protobufs -- usually we do something like:

`assert.Empty(cmp.Diff(got, tc.res, protocmp.Transform()), "GetAccount(%q) got response %q, wanted %q", tc.req, got, tc.res)` so it's using protocmp.Transform to get rid of values that are not user specified but may be distinct in the struct. But maybe it works fine? 🤷

Reply:
To be explicit could also do a specific check on the value(s) that we know will be different between two authorize calls if this _isn't_ fixed.